lib/avro/builder/dsl.rb

```diff
@@ -58,7 +58,7 @@ def type(name, type_name, options = {}, &block)
       end
 
       # Lookup an Avro schema object by name, possibly fully qualified by namespace.
-      def lookup(key, required: true)
+      def lookup(key)
```

Comment:
`type` can be moved to private now. It no longer does anything useful directly in the DSL.

I'm less worried about `lookup` since it is just a read operation. It does not make any change to the schema that is being built. It is also called by several other classes, so they would either need to use `send` or some other approach to hide it from the DSL but make it accessible to other classes.